primitives/self-contained/src/unchecked_extrinsic.rs

```diff
@@ -0,0 +1,167 @@
+use crate::{CheckedExtrinsic, CheckedSignature, SelfContainedCall};
+use codec::{Decode, Encode};
+use frame_support::{
+	traits::ExtrinsicCall,
+	weights::{DispatchInfo, GetDispatchInfo},
+};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{
+		self, Checkable, Extrinsic, ExtrinsicMetadata, IdentifyAccount, MaybeDisplay, Member,
+		SignedExtension,
+	},
+	transaction_validity::{InvalidTransaction, TransactionValidityError},
+	RuntimeDebug,
+};
+
+/// A extrinsic right from the external world. This is unchecked and so
+/// can contain a signature.
+#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo)]
+pub struct UncheckedExtrinsic<Address, Call, Signature, Extra: SignedExtension>(
+	pub sp_runtime::generic::UncheckedExtrinsic<Address, Call, Signature, Extra>,
+);
```

Comment:
This wrapping technique makes a lot of sense. I could imagine using this same technique in a repo that brings bitcoin compatibility to substrate. Same for any other chain.

If I want a chain that uses frontier and also the bitcoin-compatibility stuff, I would want to compose the unchecked extrinsic wrappers from both repos. Therefore, I think it would be cool to be generic over the thing you're wrapping.

```rust
pub struct FrontierUncheckedExtrinsic<I: Extrinsic + Checkable + ...>{
  inner: I,
}
```

Reply:
We can't yet do that now cause the struct relies on the specific information available in `UncheckedExtrinsic`, but I agree this is the direction to go.